moveos/moveos-stdlib/moveos-stdlib/sources/context.move

```diff
@@ -92,61 +95,58 @@ module moveos_std::context {
     }
 
 
-    // Wrap functions for StorageContext and ObjectRef 
+    // Wrap functions for StorageContext
 
-    #[private_generics(T)]
     /// Borrow Object from object store with object_id
-    public fun borrow_object<T: key>(self: &Context, object_id: ObjectID): &Object<T> {
-        storage_context::borrow<T>(&self.storage_context, object_id)
+    /// Any one can borrow an &ObjectRef from the global object storage
+    public fun borrow_object<T: key>(self: &Context, object_id: ObjectID): &ObjectRef<T> {
+        let object_entity = storage_context::borrow<T>(&self.storage_context, object_id);
+        object_ref::as_ref(object_entity)
     }
 
-    #[private_generics(T)]
     /// Borrow mut Object from object store with object_id
-    public fun borrow_object_mut<T: key>(self: &mut Context, object_id: ObjectID): &mut Object<T> {
-        storage_context::borrow_mut<T>(&mut self.storage_context, object_id)
+    /// Only the owner can borrow an &mut ObjectRef from the global object storage
+    public fun borrow_object_mut<T: key>(self: &mut Context, owner: &signer, object_id: ObjectID): &mut ObjectRef<T> {
+        let object_entity = storage_context::borrow_mut<T>(&mut self.storage_context, object_id);
+        let owner_address = signer::address_of(owner);
+        assert!(object::owner(object_entity) == owner_address, error::permission_denied(ErrorObjectOwnerNotMatch));
+        object_ref::as_mut_ref(object_entity)
     }
 
     #[private_generics(T)]
-    /// Remove object from object store, and unpack the Object
-    public fun remove_object<T: key>(self: &mut Context, object_id: ObjectID): (ObjectID, address, T) {
-        let obj = storage_context::remove<T>(&mut self.storage_context, object_id);
-        object::unpack_internal(obj)
+    /// The module of T can borrow mut Object from object store with any object_id
+    public fun borrow_object_mut_extend<T: key>(self: &mut Context, object_id: ObjectID) : &mut ObjectRef<T> {
```

Comment:
If an object is transferred to another address, the module where the object<T> is defined can still get the write reference of the object through `borrow_object_mut_extend`?

Reply:
yes